### ISSUE TITLE
Class View flickers as code is edited

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#27183]
When editing a C# class with the Class View open, every single change to the class file (that is, every key press that results in a text change) causes the lower pane of the Class View to flicker. The flicker appears to be of the &quot;clear pane and redraw identical content without double-buffering&quot; type.  It can be quite distracting.  Presumably behind the scenes Studio is analyzing the class file after every change, coming up with a new list of members, and redrawing the lower pane - regardless of whether the editing has actually resulted in a change to this list. There is a less noticeable flicker in the upper pane as well, as the name of the current class is apparently redrawn (as well as other random class names, occasionally). I can reproduce the problem with any C# class, but the flicker is more pronounced for classes that have a long list of members.

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/169281/class-view-flickers-as-code-is-edited.html
VSTS ticketId: 540947_
_These are the original issue comments:_

Ella Zhao (MSFT) on 12/27/2017, 07:23 PM (149 days ago): <p>Thank you for your feedback! </p><p>For us to investigate this further, we are requesting additional information outlined below. </p><ol><li>Does this repro for just C# project or any project?</li><li>Does this repro for a specific programming language or any programming language?</li></ol><p>We look forward to hearing from you!</p>

Greg Lesher on 12/29/2017, 05:36 AM (147 days ago): <p>Problem occurs for all C# projects I've tried.  I don't generally program in other languages, but just verified that the problem also occurs with Visual Basic (although perhaps with slightly less flickering) but does not appear to occur with Visual C++.</p>

Lucia Qiao [MSFT] on 1/1/2018, 10:48 PM (144 days ago): <p>Thank you for your reply Greg, And so sorry I did not repro this issue with the description you provided. So could you please provide a video of this issue for us? Or could you please provide some more detail/specified steps with screenshots instead? Thank you again, we're looking forward your reply.</p>
Greg Lesher on 1/2/2018, 06:57 AM (143 days ago): Here's some recorded video that goes with the step-by-step guide to reproducing the problem.
Lucia Qiao [MSFT] on 1/1/2018, 10:48 PM (144 days ago): <p>Thank you for your reply Greg, And so sorry I did not repro this issue with the description you provided. So could you please provide a video of this issue for us? Or could you please provide some more detail/specified steps with screenshots instead? Thank you again, we're looking forward your reply.</p>
Greg Lesher on 1/2/2018, 06:32 AM (143 days ago): Steps to reproduce (now repeated with 15.2.2 on Windows 7 and Windows 10):1) Create new Visual C# project.  I chose Console App from Windows Classic Desktop with .NET Framework 4.6.1, but every different app type shows the same problem.2) Open a Class View window and browse within the top half of the Class View window so that the Program class is selected (and thus the Program class members are shown in the lower half of the Class View window).  I have the Class View window situated to the left of the code editing window, but the problem reproduces regardless of the location.3) In the Program.cs editing window, start typing anything anywhere.  For example, type a comment in the line before the static constructor.  Because there aren't many class members for Program, flickering will be minimal in lower half of Class View window, but it should be apparent after every few keypresses (or every keypress if you type slowly).  Sometimes the flicker includes all of the class members, sometimes just the last entry (which is ToString()).4) To exacerbate the problem, create a bunch of empty functions public void Test1(), public void Test2(), etc.  Then add a comment somewhere and watch for flicker.I'll include a screenshot of Studio after Step 4.  I'll also try to generate a diagnostic video through Studio, although I suspect that the flickering will not appear because it's dependent on the timing between the redraw and the screen refresh.
Greg Lesher on 1/2/2018, 06:53 AM (143 days ago): <p>I apologize for the formatting disaster of the previous message.  Let's try again...</p><p>Steps to reproduce (now repeated with 15.2.2 on Windows 7 and Windows 10): </p><p>1) Create new Visual C# project.  I chose Console App from Windows Classic Desktop with .NET Framework 4.6.1, but every different app type shows the same problem. </p><p>2) Open a Class View window and browse within the top half of the Class View window so that the Program class is selected (and thus the Program class members are shown in the lower half of the Class View window).  I have the Class View window situated to the left of the code editing window, but the problem reproduces regardless of the location. </p><p>3) In the Program.cs editing window, start typing anything anywhere.  For example, type a comment in the line before the static constructor.  Because there aren't many class members for Program, flickering will be minimal in lower half of Class View window, but it should be apparent after every few keypresses (or every keypress if you type slowly).  Sometimes the flicker includes all of the class members, sometimes just the last entry (which is ToString()). </p><p>4) To exacerbate the problem, create a bunch of empty functions public void Test1(), public void Test2(), etc.  Then add a comment somewhere and watch for flicker.</p><p>I've included a screenshot of Studio after Step 4.</p><p>I'll also try to include a video generated through Studio, although I fear that the flickering will not appear because it's dependent on the timing between the redraw and the screen refresh.</p>

_These are the original issue solutions:_
(no solutions)

**Comment by cyraid** [comment]
Oh here's where this bug is.. Any updates or workarounds?

**Comment by mahmoodkhan** [comment]
My screen just started flickering last week after updating my VSCode. I used it for a Ruby project on a Mac OS Mojave 10.14.3 (18D42).
My VsCode version is: Version 1.31.0 (1.31.0)

It flickers only when I am connected to an external 4k display monitor.

**Comment by cyraid** [comment]
This bug report is against visual studio 2017 in the class view tool window. :) (Not vscode).

You might want to open a bug report on that project? Hope you get that issue sorted out!

**Comment by EmKayUltra** [comment]
My issue might be related... Visual Studio 2017 (tried various versions) is randomly causing not just flicker within the IDE, but on my entire screen (laptop).  It'll flicker sometimes multiples times per minute, sometimes less.  It inevitably concludes with my screen going entirely blank or something much worse (see [here](https://imgur.com/a/B58qsQh) - this level of display corruption has only occurred once, just now).  

Using the power button to put the machine to sleep and immediately wake it up brings everything back to normal for a little bit, until it happens again.  

It's definitely related to Visual Studio 2017, as it has not occurred once in hundreds of hours of using my machine without VS 2017.

**Comment by CyrusNajmabadi** [comment]
@EmKayUltra Do you have any of these options checked:

![image](https://user-images.githubusercontent.com/4564579/53601720-96d63d00-3b61-11e9-876b-e363ceab6c6b.png)

?

**Comment by EmKayUltra** [comment]
@CyrusNajmabadi  I tried unchecking both, it didn't help.  I actually think this may be project-specific (maybe related to Library Manager), as I opened up a different project, which doesn't use Library Manager, and haven't had a single problem for the last 30 minutes (which is a long time for me at this point).

**Comment by CyrusNajmabadi** [comment]
>  I tried unchecking both, it didn't help. I actually think this may be project-specific (maybe related to Library Manager), as I opened up a different project, which doesn't use Library Manager, and haven't had a single problem for the last 30 minutes (which is a long time for me at this point)

Fair enough.  Note that LibraryManager is now owned by roslyn.  if you're having issues with it, you'll have to open a general VS issue here:

![image](https://user-images.githubusercontent.com/4564579/53602888-e0745700-3b64-11e9-9ac9-d273461e11f3.png)

**Comment by EmKayUltra** [comment]
Roger that.  I'm not sure it IS libman, though.

**Comment by geoff-m** [comment]
> Fair enough. Note that LibraryManager is now owned by roslyn. if you're having issues with it, you'll have to open a general VS issue here:

I don't know what LibraryManager is but the report of this issue on that I made through "Report a Problem..." was closed three days later as a duplicate of https://developercommunity.visualstudio.com/content/problem/169281/class-view-flickers-as-code-is-edited.html, which links in turn to this github issue.

**Comment by jinujoseph** [comment]
@olegtk Does editor own this ?

**Comment by olegtk** [comment]
Yes, it's owned by the editor team, please route it.

**Comment by CyrusNajmabadi** [comment]
Closing out as this is owned by editor.
